### PR TITLE
Misc bugfixes to trigger -> build -> backend chain

### PR DESCRIPTION
**Pull request by Berlioz**
Addresses some problems I noticed while testing locally:
1) To get the default serviceAccountEmail, you set the field to null in backend.backend, not "default"
2) I forgot about SecretEnvVariables entirely
3) None of this makes any sense in the context of CF3v2 (the code in runtimes/discovery should take care of that, right?)

### CODE REVIEW

File: src/deploy/functions/prepare.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ export async function prepare(
     const userEnvs = functionsEnv.loadUserEnvs(userEnvOpt);
     const envs = { ...userEnvs, ...firebaseEnvs };
     let wantBackend: backend.Backend;
-    if (previews.functionsparams) {
+    if (previews.functionsparams && !previews.functionsv2) {
```
Comment (inlined) [inline src/deploy/functions/prepare.ts:117]: previews.functionsv2 is gone now.

File: src/deploy/functions/runtimes/node/parseTriggers.ts [hunk 2]
```diff
@@ -302,7 +302,7 @@ export function addResourcesToBuild(
     project: projectId,
     entryPoint: annotation.entryPoint,
     runtime: runtime,
-    serviceAccount: annotation.serviceAccountEmail || "default",
+    serviceAccount: annotation.serviceAccountEmail || null,
```
Comment (inlined) [inline src/deploy/functions/runtimes/node/parseTriggers.ts:305]: It's quite possible that `undefined` will trigger different behavior than `null`. They behave differently in `JSON.stringify` and `proto.fieldMasks` includes nulls but not undefined too. This likely has the side effect of resetting service accounts on every deploy to their default.

**Review by inlined — APPROVED** [review]
Sorry I got so caught up in I/O stuff that I missed this. If you ever wait more than a full day for a review, please ping me.

**Comment by inlined** [comment]
This has been sitting around for about a month; should this be committed or closed?

**Comment by Berlioz** [comment]
We're closing this one; the relevant changes in here have been folded into other PRs.

**Pull request closed without merging**